### PR TITLE
BlobDetection

### LINKED ISSUE #2306: Detect blobs in a volume

### PR DESCRIPTION
**Pull request by mcquin**
Resolves #2306

### CODE REVIEW

**Comment by mcquin** [comment]
Will rebase before merge.

**Comment by 0x00b1** [comment]
Would you mind posting a screenshot?

File: cellprofiler/modules/blobdetection.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
+# coding=utf-8
+
+"""
+
+Find blobs in a grayscale (single-channel) image or volume. Blobs are bright on dark or dark on bright regions in an
```
Comment (0x00b1) [inline cellprofiler/modules/blobdetection.py]: Are you saying that it’s limited to single-channel two-dimensional and single-channel multi-dimensional images? Is there a reason this cannot be extended to multichannel two-dimensional and multichannel three-dimensional images? Is it possible to do so by pre-processing with `rgb2gray`?

**Comment by mcquin** [comment]
@0x00b1 here's one https://github.com/CellProfiler/CellProfiler/issues/2306

File: cellprofiler/modules/blobdetection.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+image. The output of this module is a binary image of white circles or spheres centered around found blobs on a black
```
Comment (0x00b1) [inline cellprofiler/modules/blobdetection.py]: Nice. 😎

I assumed it’d return boundaries, but filled circles are probably more useful.

File: cellprofiler/modules/blobdetection.py [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+        ]
+
+        if self.operation.value in [u"Determinant of the Hessian (DoH)", u"Laplacian of Gaussian (LoG)"]:
+            settings = settings + [
+                self.scale
+            ]
+
+        return settings
+
+    def on_setting_changed(self, setting, pipeline):
```
Comment (0x00b1) [inline cellprofiler/modules/blobdetection.py]: What is this?

**Comment by 0x00b1** [comment]
Awesome work

File: cellprofiler/modules/blobdetection.py (hunk 1 above)
Comment (mcquin) [inline cellprofiler/modules/blobdetection.py]: I thought about using `rgb2gray`, but that assumes all multi-channel images are RGB images.

File: cellprofiler/modules/blobdetection.py (hunk 3 above)
Comment (mcquin) [inline cellprofiler/modules/blobdetection.py]: Each operation's `skimage.feature` equivalent has slightly different default parameters. This has the benefit of changing the settings to match those default parameters when the operation is changed. The downside is that it forgets/wipes out user defined settings when the operation is changed.

**Comment by mcquin** [comment]
Squashed.

**Pull request merged**